src/components/provisionerForm/SSHKeys.js

```diff
@@ -31,88 +49,110 @@ const SSHKeys = ( {deleteSSHKey, enableSSHKey, setSSHKey, setSSHKeyTitle, setSSH
     }));
   };
   
-  return (
-    <div
-      className="row"
-      data-magellan-destination="ssh-keys" 
-      id="ssh-keys">
-    <h2>
-      <i className="step fi-key"></i>
-       Add SSH Keys
-    </h2>
-    <div id="ssh_keys">
-    </div>
-      <div className="large-12 medium-12 small-12 columns end">
-        <p><a
-          href="javascript:void(0);"
-          id="show_ssh_key">+ Add SSH Key</a>
-        </p>
-        <div className="ssh_key_content">
-          <div className="large-offset-9 large-3 medium-9">
-            <p className="right">
-              <a
-                href="javascript:void(0);"
-                id="hide_ssh_key">
-                Close</a>
-            </p>
-          </div>
-          <form>
-            <div className="row">
-              <div className="columns">
-                <label className="error">
-                  <textarea
-                    id="ssh_key_content_value"
-                    cols="50"
-                    rows="5"
-                    onChange={AddSSHKeyContent}
-                    placeholder="SSH Key Content"></textarea>
-                </label>
-                <div 
-                  className="hide" 
-                  id="ssh_key_content_value_error">
-                  <small className="error hide">SSH Key Content can't be blank</small>
-                </div>
+  const AddedSSHKeys = 
+    (sshKeysAppState.get('ssh_keys')) ?
+      sshKeysAppState.get('ssh_keys').map((value, index) => 
+        <SSHKeysItem
+          handleClick={handleSSHKeyClick}
+          key={index}
+          value={value.get('title')}
+          isActive={value.get('enable')?true:false}
+          end={(index == sshKeysAppState.get('ssh_keys').size - 1) ? "end" : ""}
+        />) : "";
+  
+  const AddSHHKey = 
+    !(sshKeysAppState.get('show_ssh_key')) ?
+      <a
+        href="javascript:void(0);"
+        onClick={ShowSSHKeyKeypress}
+        id="show_ssh_key">+ Add SSH Key</a> : "";
+      
+  const ShowSSHKey =
```

Comment:
This seems like a rather large method, maybe can be in a component or something, up to you.